packages/big-design/src/components/Counter/spec.tsx

```diff
@@ -1,320 +1,335 @@
-  const counter = getByDisplayValue('5') as HTMLInputElement;
+  const counter = screen.getByRole<HTMLInputElement>('textbox');
   counter.focus();
 
   expect(counter.value).toEqual('5');
-  fireEvent.keyDown(counter, { key: 'ArrowUp', code: 'ArrowUp' });
+  userEvent.type(counter, '{arrowup}');
   expect(handleChange).toHaveBeenCalledWith(6);
-  fireEvent.keyDown(counter, { key: 'ArrowDown', code: 'ArrowDown' });
+  userEvent.type(counter, '{arrowdown}');
   expect(handleChange).toHaveBeenCalledWith(4);
 });
 
 test('value is set to 0 when pressing Escape', () => {
-  const { getByDisplayValue } = render(counterMock(requiredAttributes));
+  const handleChange = jest.fn((num: number) => num);
+  render(<CounterMock value={5} onCountChange={handleChange} />);
 
-  const counter = getByDisplayValue('5');
+  const counter = screen.getByRole<HTMLInputElement>('textbox');
 
-  expect(counter.getAttribute('value')).toEqual('5');
-  fireEvent.keyDown(counter, { key: 'Escape', code: 'Escape' });
+  expect(counter).toHaveValue('5');
+  userEvent.type(counter, '{esc}');
   expect(handleChange).toHaveBeenCalledWith(0);
 });
 
 test('value does not change when pressing Enter', () => {
-  const { getByDisplayValue } = render(counterMock(requiredAttributes));
+  const handleChange = jest.fn((num: number) => num);
+  render(<CounterMock value={5} onCountChange={handleChange} />);
 
-  const counter = getByDisplayValue('5');
+  const counter = screen.getByRole<HTMLInputElement>('textbox');
 
-  expect(counter.getAttribute('value')).toEqual('5');
-  fireEvent.keyDown(counter, { key: 'Enter', code: 'Enter' });
+  expect(counter).toHaveValue('5');
+  userEvent.type(counter, '{enter}');
   expect(handleChange).not.toHaveBeenCalled();
 });
 
 test('provided onCountChange function is called on value change', () => {
-  const { getByTitle } = render(counterMock(requiredAttributes));
+  const handleChange = jest.fn((num: number) => num);
+  render(<CounterMock value={5} onCountChange={handleChange} />);
 
-  const increase = getByTitle('Increase count');
+  const increase = screen.getByTitle('Increase count');
 
-  fireEvent.click(increase);
+  userEvent.click(increase);
   expect(handleChange).toHaveBeenCalledWith(6);
 });
 
 test('error shows with valid string', () => {
   const error = 'Error';
-  const { container, rerender } = render(<FormGroup>{counterMock({ error: '', ...requiredAttributes })}</FormGroup>);
+  const { rerender } = render(<FormGroup>{<CounterMock value={5} error="" />}</FormGroup>);
 
-  expect(container.querySelector('[class*="StyledError"]')).not.toBeInTheDocument();
+  expect(screen.queryByText('Error')).not.toBeInTheDocument();
 
-  rerender(<FormGroup>{counterMock({ error, ...requiredAttributes })}</FormGroup>);
+  rerender(<FormGroup>{<CounterMock value={5} error={error} />}</FormGroup>);
 
-  expect(container.querySelector('[class*="StyledError"]')).toBeInTheDocument();
+  expect(screen.getByText(error).parentElement).toBeInTheDocument();
```

Comment:
If `error` is in the document, `parentElement` should also be in the document, is it really necessary to go up one level?